### ISSUE TITLE
[Bug]: Movement blocking layer appears to be broken in MapTool 1.11.4

### ISSUE THREAD

**Issue by adventuremagic123** [RPTools/maptool#3289]
### Describe the Bug

My players can cross the movement blocking layer with their tokens.

### To Reproduce

1. Start MapTool 1.11.4 and (important) start the server.
2. Make sure the server is started (this is a work-around for another issue -- must be done first) and load the following example campaign:

[Blocking layer example.zip](https://github.com/RPTools/maptool/files/7738622/Blocking.layer.example.zip)

3. Ensure the AI option is turned off.
4. Drag the token across the blocking layers as the player who owns the token (you'll have to setup the owner of the token).
5. Notice that the movement blocking layer does NOT stop the token as in the following image:

![MBL appears broken](https://user-images.githubusercontent.com/50708405/146626910-c8eaa513-3897-4023-91fb-4c4ee9857b4b.jpg)

### Expected Behaviour

I'm expecting that players should not be able to move their tokens across the movement blocking layer.  GMs should be able to, but not players.

### MapTool Info

1.11.4

### Desktop

Windows 10

**Comment by Azhrei** [comment]
Movement Blocking only affects AI path routing.  Turning off the AI means the MT will not impose any restrictions on your movement.  (Probably should call it something else, like Enforce Movement Penalties…?)

**Comment by adventuremagic123** [comment]
I didn't want to have to use AI routing.  Previously, I was able to turn it off and I thought MBL worked.

**Comment by adventuremagic123** [comment]
AI routing is sometimes too revealing.

**Comment by Phergus** [comment]
As Azhrei said,  MBL only affects AI pathfinding.  It does not and never has actually blocked movement.  This is intentional.  A character that flies would not be inconvenienced by a chasm or fire pit and could just go directly over it.  The controlling player needs to be able to make that movement.

**Comment by adventuremagic123** [comment]
lib:elevation (a library token by Melek) is a better way to address the flying problem.  In older MapTool versions before MBL was introduced, VBL blocked movement without using AI routing.  Later, MBL was introduced and I have been using it to block movement without AI routing.  I'm planning to demonstrate this later today.

**Comment by adventuremagic123** [comment]
For me, this is a show stopper issue.

**Comment by Phergus** [comment]
> In older MapTool versions before MBL was introduced, VBL blocked movement without using AI routing.

VBL does not and never has blocked movement.  Unexposed areas of FoW do block movement which may give the impression that it is the VBL blocking movement.

Nothing has changed in regards to MBL & VBL and their effect on movement.

**Comment by adventuremagic123** [comment]
How can you say that for MapTool 1.4.x, for example?

**Comment by Phergus** [comment]
I can say this because I've been working with MapTool for 15 years.   I have 43 different versions of MapTool installed currently going back to 1.1.b35.

**Comment by adventuremagic123** [comment]
I've been using it that way for 3 years.

**Comment by adventuremagic123** [comment]
I'm investigating.  It appears the behavior Phergus and Azhrei describe has been there since maybe AI was introduced.  I've looked back to MapTool 1.9.3 so far.  I can't use it this way.

**Comment by Phergus** [comment]
Here is a video showing that it worked exactly as we've described back in MT 1.4.0.5.  This is pre-AI.

https://www.dropbox.com/s/049jzlifhmdqafc/VBL-FoW-Example_MT1405.mp4?dl=0

**Comment by adventuremagic123** [comment]
Yes, I guess I just completely misunderstood how it works.  Thanks for the video.  I don't know how that's useful that way.  Please don't tell me that's how it works in D20Pro and Foundry VTT.

**Comment by Azhrei** [comment]
What is it you're trying to accomplish?  Since you're starting out by turning off the AI button, I guess the question would be, why not just leave it on?

**Comment by adventuremagic123** [comment]
I explained that above.  I don't like the AI feature.  I like the blocking layer behavior I see in Foundry VTT.  I just did a test with Foundry VTT 0.8.9.  I'll bet D20Pro does the same thing, but I haven't tested it yet.  I came from D20Pro and apparently assumed MapTool behaved the same way.

**Comment by adventuremagic123** [comment]
> AI routing is sometimes too revealing. I don't think it respects FoW or vision limitations. A less common use case is what if I use VBL to conceal a passage way. AI routing screams out, "Here it is." I typically do not like to use AI routing.

As I said above ...

AI routing is sometimes too revealing. I don't think it respects FoW or vision limitations. A less common use case is what if I use VBL to conceal a passage way. AI routing screams out, "Here it is." I typically do not like to use AI routing.

**Comment by adventuremagic123** [comment]
I'm sorry.  I'm kind of upset about this.  I can't see how I can use this.  I spent a lot of time developing my framework, but there's just no way I can make this work.

**Comment by Phergus** [comment]
MapTool has worked the same way the entire time you have been using it.

You still haven't adequately explained what it is you are trying to accomplish.

**Comment by adventuremagic123** [comment]
I want the MBL to stop players from dragging their tokens across it.  That's how it works in other VTTs.

**Comment by aliasmask** [comment]
Perhaps there's something with the AI mode that needs to be addressed. What does it do that you don't like. I know you mentioned revealing secret doors somehow.

**Comment by adventuremagic123** [comment]
It reveals pathways that players shouldn't necessarily know about.  When they flick their mouse around, it reveals all the possible passage ways their token could follow -- and, thus, kind of eliminates part of the mystery part of FoW.

It's much more fun to have players slowly venture forward with their tokens than to have something that says without revealing FoW that there's a passage here and here and here and here and so on ...

**Comment by adventuremagic123** [comment]
I'm going to post a feature request to add an option to get the behavior I would like to have to run my campaigns.

**Comment by adventuremagic123** [comment]
I apologize to the community for my emotionally charged comments.  We're going through a difficult personal time.  Since May, it's really this community that's helped a lot with keeping me on my feet.  You're all my friends.
